src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs

```diff
@@ -2,84 +2,114 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Help;
 using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.Extensions.CommandLineUtils;
 using NuGet.Configuration;
 using NuGet.Credentials;
 
 namespace NuGet.CommandLine.XPlat
 {
     internal class PackageSearchCommand
     {
-        public static void Register(CommandLineApplication app, Func<ILoggerWithColor> getLogger)
+        public static void Register(CliCommand rootCommand, Func<ILoggerWithColor> getLogger)
         {
-            Register(app, getLogger, SetupSettingsAndRunSearchAsync);
+            Register(rootCommand, getLogger, SetupSettingsAndRunSearchAsync);
         }
 
-        public static void Register(CommandLineApplication app, Func<ILoggerWithColor> getLogger, Func<PackageSearchArgs, Task<int>> setupSettingsAndRunSearchAsync)
+        public static void Register(CliCommand rootCommand, Func<ILoggerWithColor> getLogger, Func<PackageSearchArgs, CancellationToken, Task<int>> setupSettingsAndRunSearchAsync)
         {
-            app.Command("search", pkgSearch =>
+            var searchCommand = new CliCommand("search", Strings.pkgSearch_Description);
+
+            var searchTerm = new CliArgument<string>("Search Term")
+            {
+                Description = Strings.pkgSearch_Description,
```

Comment:
Should this be pkgSearch_termDescription?

Reply:
Nice catch! Fixed.